File: easy/Linked Lists/Reverse Linked List.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if (!head || !head->next) return head;

        std::vector<int> vals;
        ListNode* x = head;

        while (x) {
            vals.push_back(x->val);
            x = x->next;
        }

        x = head;
        for (int i = vals.size() - 1; i >= 0; --i) {
            x->val = vals[i];
            x = x->next;
        }

        return head;
    }
};

File: easy/Linked Lists/Reverse Linked List.java
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) return head;

        ArrayList<Integer> list = new ArrayList<>();
        ListNode x = head;

        while (x != null) {
            list.add(x.val);
            x = x.next;
        }

        x = head;
        int i = list.size() - 1;
        while (x != null) {
            x.val = list.get(i--);
            x = x.next;
        }

        return head;
    }
}
